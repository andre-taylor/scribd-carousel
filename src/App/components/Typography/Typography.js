import React from 'react';
import PropTypes from 'prop-types';

export const Typography = ({ variant, className, content, children }) => {
  return React.createElement(variant, { className }, children || content);
};

Typography.defaultProps = {
  variant: 'span',
};

Typography.propTypes = {
  variant: PropTypes.oneOf(['h2', 'h3', 'h4', 'span', 'p']).isRequired,
  className: PropTypes.string,
  content: PropTypes.node,
  children: function (props, propName) {
    // children is opitional as long as prop.content has a value
    if (props[propName] === undefined && props['content'] === undefined) {
      return new Error('Component requires a content prop value!');
    }
  },
};

export default Typography;
