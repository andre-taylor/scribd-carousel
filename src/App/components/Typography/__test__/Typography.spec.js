import React from 'react';
import renderer from 'react-test-renderer';

import Typography from '../Typography';

describe('<Typography />', () => {
  it('renders correctly with default props and element children props', () => {
    const component = renderer.create(
      <Typography>The quick brown fox jumped over the lazy dog.</Typography>
    );
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with default and content props', () => {
    const component = renderer.create(
      <Typography content={'The quick brown fox jumped over the lazy dog.'} />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with a "h2" variant', () => {
    const component = renderer.create(
      <Typography
        variant={'h2'}
        className={'cssStyle'}
        content={'The quick brown fox jumped over the lazy dog.'}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with a "h3" variant', () => {
    const component = renderer.create(
      <Typography
        variant={'h3'}
        className={'cssStyle'}
        content={'The quick brown fox jumped over the lazy dog.'}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with a "h4" variant', () => {
    const component = renderer.create(
      <Typography
        variant={'h4'}
        className={'cssStyle'}
        content={'The quick brown fox jumped over the lazy dog.'}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with a "span" variant', () => {
    const component = renderer.create(
      <Typography
        variant={'span'}
        className={'cssStyle'}
        content={'The quick brown fox jumped over the lazy dog.'}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders correctly with a "p" variant', () => {
    const component = renderer.create(
      <Typography
        variant={'p'}
        className={'cssStyle'}
        content={'The quick brown fox jumped over the lazy dog.'}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
