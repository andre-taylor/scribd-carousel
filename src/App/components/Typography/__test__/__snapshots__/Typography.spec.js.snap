// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> renders correctly with a "h2" variant 1`] = `
<h2
  className="cssStyle"
>
  The quick brown fox jumped over the lazy dog.
</h2>
`;

exports[`<Typography /> renders correctly with a "h3" variant 1`] = `
<h3
  className="cssStyle"
>
  The quick brown fox jumped over the lazy dog.
</h3>
`;

exports[`<Typography /> renders correctly with a "h4" variant 1`] = `
<h4
  className="cssStyle"
>
  The quick brown fox jumped over the lazy dog.
</h4>
`;

exports[`<Typography /> renders correctly with a "p" variant 1`] = `
<p
  className="cssStyle"
>
  The quick brown fox jumped over the lazy dog.
</p>
`;

exports[`<Typography /> renders correctly with a "span" variant 1`] = `
<span
  className="cssStyle"
>
  The quick brown fox jumped over the lazy dog.
</span>
`;

exports[`<Typography /> renders correctly with default and content props 1`] = `
<span>
  The quick brown fox jumped over the lazy dog.
</span>
`;

exports[`<Typography /> renders correctly with default props and element children props 1`] = `
<span>
  The quick brown fox jumped over the lazy dog.
</span>
`;
